omegaconf/_utils.py

```diff
@@ -491,7 +491,7 @@ def valid_value_annotation_type(type_: Any) -> bool:
 
 
 def _valid_dict_key_annotation_type(type_: Any) -> bool:
-    return type_ is None or type_ is Any or issubclass(type_, (str, Enum))
+    return type_ is None or type_ is Any or issubclass(type_, (str, int, Enum))
```

Comment:
I think a `DictKeyType = Union[str, int, Enum]` should be defined somewhere so that whenever it's changed we only need to modify it in one place.

Here this would translate into writing `issubclass(type_, DictKeyType.__args__)`

Edit: just saw that @omry had the same comment, still leaving this one to show how to obtain the underlying types.